src/components/molecules/LocationMap.tsx

```diff
@@ -1,32 +1,66 @@
 import 'leaflet-css'
 import React, { FunctionComponent } from 'react'
 import L from 'leaflet'
-import { Map, Marker, TileLayer } from 'react-leaflet'
+import { Map, TileLayer } from 'react-leaflet'
+import AnywayMarker from '../atoms/AnywayMarker'
 
 // TODO: Move to init.service once it's merged.
 L.Icon.Default.mergeOptions({
     iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
     iconUrl: require('leaflet/dist/images/marker-icon.png'),
     shadowUrl: require('leaflet/dist/images/marker-shadow.png')
 })
-
 interface IProps {
+    data: any[],
     marker: { lat: number; lng: number }
 }
-
 const INITIAL_ZOOM = 13
 const WRAPPER_STYLES = { height: '100%', width: '100%' }
 
-const LocationMap: FunctionComponent<IProps> = ({ marker }) => {
+const LocationMap: FunctionComponent<IProps> = ({ data, marker }) => {
+    let markers = data.map((x: any) => {
+        if (x.latitude !== null && x.longitude !== null) {
+            return <AnywayMarker markerdata={x} />
+        }
+        else return null;
+    });
+    const bounds = setBounds(data)
+
     return (
-        <Map center={marker} zoom={INITIAL_ZOOM} style={WRAPPER_STYLES}>
+        <Map center={marker} bounds={bounds} zoom={INITIAL_ZOOM} style={WRAPPER_STYLES}>
             <TileLayer
                 url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
                 attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
             />
-            <Marker position={marker} />
+            {markers}
         </Map>
     )
 }
-
+const setBounds = (data: any[]) => {
+    const corner1 = L.latLng(29.50, 34.22)
+    const corner2 = L.latLng(33.271, 35.946)
+    let arr: L.LatLng[] = [];
+    let lastPoint: L.LatLng = L.latLng(0, 0);
+    data.forEach(x => {
+        if (x.latitude !== null && x.longitude !== null) {
+            let p = new L.LatLng(x.latitude, x.longitude);
```

Comment:
You can create this inside the `if` block, as you only use it there. This would save unnecessary points creations.

Reply:
in need the lastpoint in case there is only single point , maybe arr[arr.length -1] will also work...